specs/tables.md

```diff
@@ -41,14 +41,14 @@ NOTE:
 Proved by the withdrawal circuit.
 
 This circuit 
-| 0 Withdrawal ID | 1 Validator ID | 2 Address      | 3 Amount      | 4 MPT root     |
```

Comment:
From our discussion, we'll keep mpt root, right?

Reply:
Just to double check: Here mpt root is meant `intermediate_mpt_root` right, and keep intermediate_mpt_root used it during lookup to assure MPT insertion sequential? 

I saw all MPT lookup in withdrawl root only provide `key` information, is it enough to assure MPT root insertion maniputation are indeed [insert wd1, insert wd2, ..., insert wdN] without involving extra node in MPT table ?

Reply:
For @KimiWu123,
> From our discussion, we'll keep mpt root, right?

We don't need to keep mpt roots in the table, because we don't need them in the EVM circuit. 

We do need to keep mpt roots in the withdrawal circuit. Let me summarize our discussion for future reference.

Say we have a tree of 3 withdrawals `{0: wd0, 1: wd1, 2: wd2}` and its root `withdrawal_root`.

The withdrawal circuit should prove the validity of the withdrawal table and the valid table should have `[wd0, wd1, wd2]`.

It is not enough to show `membership_proof(withdrawal_root, 0, wd0)`, `membership_proof(withdrawal_root, 1, wd1)`, and `membership_proof(withdrawal_root, 2, wd2)` in the withdrawal circuit, because prover might skip the last membership proof in the circuit and remove `wd2` from the withdrawal table. 

The sound withdrawal circuit should build the tree incrementally. It starts with empty root `{}`, then insert wd0 to get `{0: wd0}`, then wd1 for `{0: wd0, 1: wd1}`, and finally `{0: wd0, 1: wd1, 2: wd2}`. It shows that the final tree has the tree root `withdrawal_root`. By constructing the circuit this way, a malicious prover can not skip any withdrawal data in the circuit and in the table.